src/stateStream.spec.ts

```diff
@@ -0,0 +1,134 @@
+);
+const test = stubRethrowErrorGlobally(untypedTest);
+
+const actions = {
+  '1': incrementOne(),
+  '2': incrementMany(2),
+  d: decrement(),
+};
+const words = {
+  a: '1',
+  b: '12',
+};
+const numbers = {
+  '1': 1,
+  '2': 2,
+  '3': 3,
+  '4': 4,
+  '5': 5,
+  '6': 6,
+  '7': 7,
+};
+
+test(
+  'stateStream subscribes action$',
+  marbles(m => {
+    const defaultState = 3;
+    const action$ = m.hot('', actions);
+    const state$ = stateStream('testStream', action$, defaultState, []);
+    state$.subscribe();
+    m.expect(action$).toHaveSubscriptions(['^']);
+  })
+);
+
+test(
+  'stateStream should emit initial state synchronously upon subscription',
+  marbles(m => {
+    const defaultState = 3;
+    const action$ = m.hot('');
+    const expected$ = m.hot('3', numbers);
+    const state$ = stateStream('testStream', action$, defaultState, []);
+
+    m.expect(state$).toBeObservable(expected$);
+  })
+);
+
+test(
+  'stateStream should call reducers synchronously if they emit at once',
+  marbles(m => {
+    const action$ = m.hot('', actions);
+    const word$ = m.hot('    a', words);
+    const expected$ = m.hot('(12)', numbers);
+    const defaultState = 1;
+
+    const handleWord = reducer(
+      word$,
+      (state: number, word) => state + word.length
+    );
+    const state$ = stateStream('testStream', action$, defaultState, [
+      handleWord,
+    ]);
+
+    m.expect(state$).toBeObservable(expected$);
+  })
+);
+
+test('stateStream subscribers should only see one emission if reducers emit synchronously', async t => {
+  const defaultState = 1;
+  const action$ = of(incrementOne());
+  const state$ = stateStream('testStream', action$, defaultState, [handleOne]);
+  t.is(await state$.toPromise(), 2);
+});
+
+test('stateStream should call reducer once when there are multiple subs', async t => {
+  const defaultState = 1;
+  incrementOneHandler.resetHistory();
+  const action$ = of(incrementOne());
+  const state$ = stateStream('testStream', action$, defaultState, [handleOne]);
+  const sub1 = state$.subscribe();
+  const sub2 = state$.subscribe();
+  t.assert(incrementOneHandler.calledOnce);
+  sub1.unsubscribe();
+  sub2.unsubscribe();
+});
+
+test('stateStream should never reset to its default state', async t => {
+  const defaultState = 1;
+  const action$ = of(incrementOne());
+  const state$ = stateStream('testStream', action$, defaultState, [handleOne]);
+  const sub1 = state$.subscribe();
+  sub1.unsubscribe();
+  const sub2Value = await state$.toPromise();
+  t.is(sub2Value, 2);
+});
+
+test(
+  'stateStream should continue emitting after errors within reducers',
+  marbles(m => {
+    const action$ = m.hot('  -22--1--1', actions);
+    const word$ = m.hot('    ---a-b', words);
+    const expected$ = m.hot('135--6--7', numbers);
+    const defaultState = 1;
+
+    const handleWord = reducer(word$, () => {
+      throw 'error';
+      // eslint-disable-next-line no-unreachable
+      return 1;
```

Comment:
```suggestion
    const handleWord = reducer(word$, (): number => {
      throw 'error';
```
I think this would work?

And this test would also be a bit more concise if we didn't use stream reducers (`handleWord`), but instead followed the style of this test: https://github.com/ardoq/rxbeach/blob/master/src/reducer.spec.ts#L95

Reply:
good suggestion, thanks 👍